src/components/ConversationSettings/LinkShareSettings.vue

```diff
@@ -164,35 +201,32 @@ export default {
 		async toggleGuests() {
 			const allowGuests = this.conversation.type !== CONVERSATION.TYPE.PUBLIC
 			this.isSaving = true
-			await this.$store.dispatch('toggleGuests', { token: this.token, allowGuests })
+			if (this.forcePasswordProtection && allowGuests) {
+				await this.togglePassword(allowGuests)
+				await this.$store.dispatch('toggleGuests', { token: this.token, allowGuests, password: this.password })
+			} else {
+				if (!allowGuests) {
+					await this.togglePassword(false)
```

Comment:
That briefly show a warning, before changing room type

@miaulalala can we maybe remove password in `RoomController::makePrivate()`?